R/edgeThresh.R

```diff
@@ -0,0 +1,461 @@
+                    v1 = sapply(lapply(list(pair[1, ], pair[2, ]), sample),
+                                function(x) x[1])
+                    vVector = as.vector(pair)
+                    v2 = vVector[!(vVector %in% v1)]
+                    revV1 = rev(v1)
+                    revV2 = rev(v2)
+
+                    # Form new edges, delete old edges
+                    del1 = paste0(pair[1, ], collapse = "|")
+                    del2 = paste0(pair[2, ], collapse = "|")
+                    del3 = paste0(rev(pair[1, ]), collapse = "|")
+                    del4 = paste0(rev(pair[2, ]), collapse = "|")
+                    mainComponent <-
+                        igraph::delete.edges(mainComponent, c(del1, del2, del3,
+                                                              del4))
+                    mainComponent <-
+                        igraph::add.edges(mainComponent,
+                                          c(v1, v2, revV1, revV2),
+                                          attr = attribs)
+
+                    # Update biDirs
+                    assign(x = as.character(v1[1]), value = v1[2], biDirs)
+                    assign(x = as.character(v1[2]), value = v1[1], biDirs)
+
+                    assign(x = as.character(v2[1]), value = v2[2], biDirs)
+                    assign(x = as.character(v2[2]), value = v2[1], biDirs)
+
+                    # Update i
+                    i <- i + 1
+                }
+            } else {
+                # ==== Unidirectional edge swap ================================
+
+                # Get edge weights from subgraph, prepare as attributes
+                eIDs = igraph::E(mainComponent, c(pair[1, ], pair[2, ]))$ID
+                eWeights = igraph::E(mainComponent)$weight[igraph::E(mainComponent)$ID %in% eIDs]
+
+                attribs = list(weight = eWeights, ID = eIDs)
+
+                # Swap endpoints
+                v1 = c(pair[1, 1], pair[2, 2])
+                v2 = c(pair[2, 1], pair[1, 2])
+
+                # Form new edges, delete old edges
+                del1 = paste0(pair[1, ], collapse = "|")
+                del2 = paste0(pair[2, ], collapse = "|")
+                mainComponent <- igraph::delete.edges(mainComponent,
+                                                      c(del1, del2))
+                mainComponent <- igraph::add.edges(mainComponent,
+                                                   c(v1, v2), attr = attribs)
+
+                # Update i
+                i <- i + 1
+            }
+        } else {
+            # Selected vertices are not all distinct, skip
+            next
+        }
+    }
+
+    return(mainComponent)
+
+}
+
+#' interpolates NA values in a vector by interpolateing from two
+#' bracketing non-missing values. If is.na(v[1]), the left-bracket is
+#' vMin, if is.na(v[length(v)]), vMax is the right bracket.
+#' @param v a numeric vector
+#' @param vMin numeric
```

Comment:
@RicardoRam This is a function made by Prof. Steipe for interpolating NA values, I simply copied it here.
It is the smallest value expected for the interpolation.